app/models/server/models/Users.js

```diff
@@ -174,10 +174,15 @@ export class Users extends Base {
 		return this.find(query);
 	}
 
-	getNextAgent(ignoreAgentId) { // TODO: Create class Agent
+	getNextAgent(ignoreAgentId, extraQuery) { // TODO: Create class Agent
+		// fetch all unavailable agents, and exclude them from the selection
+		const unavailableAgents = Promise.await(this.getUnavailableAgents(null, extraQuery)).map((u) => u.username);
```

Comment:
```suggestion
		const unavailableAgents = this.getUnavailableAgents(null, extraQuery).map((u) => u.username);
```
The getUnavailableAgents method isn't async so I don't think we need to resolve a promise here